test/io/test_websockets.py

```diff
@@ -0,0 +1,186 @@
+from autogen.io import IOWebsockets
+
+KEY_LOC = "notebook"
+OAI_CONFIG_LIST = "OAI_CONFIG_LIST"
+
+# Check if the websockets module is available
+try:
+    from websockets.sync.client import connect as ws_connect
+except ImportError:  # pragma: no cover
+    skip_test = True
+    skip_openai = True  # noqa: F811
+else:
+    skip_test = False
+
+    try:
+        import openai
+    except ImportError:
+        skip_openai = True  # noqa: F811
+
+
+class TestConsoleIOWithWebsockets:
+    @pytest.mark.skipif(skip_test, reason="websockets module is not available")
+    def test_input_print(self) -> None:
+        print()
+        print("Testing input/print", flush=True)
+
+        def on_connect(iostream: IOWebsockets) -> None:
+            print(f" - on_connect(): Connected to client using IOWebsockets {iostream}", flush=True)
+
+            print(" - on_connect(): Receiving message from client.", flush=True)
+
+            msg = iostream.input()
+
+            print(f" - on_connect(): Received message '{msg}' from client.", flush=True)
+
+            assert msg == "Hello world!"
+
+            for msg in ["Hello, World!", "Over and out!"]:
+                print(f" - on_connect(): Sending message '{msg}' to client.", flush=True)
+
+                iostream.print(msg)
+
+            print(" - on_connect(): Receiving message from client.", flush=True)
+
+            msg = iostream.input("May I?")
+
+            print(f" - on_connect(): Received message '{msg}' from client.", flush=True)
+            assert msg == "Yes"
+
+            return
+
+        with IOWebsockets.run_server_in_thread(on_connect=on_connect, port=8765) as uri:
+            print(f" - test_setup() with websocket server running on {uri}.", flush=True)
+
+            with ws_connect(uri) as websocket:
+                print(f" - Connected to server on {uri}", flush=True)
+
+                print(" - Sending message to server.", flush=True)
+                websocket.send("Hello world!")
+
+                for expected in ["Hello, World!", "Over and out!", "May I?"]:
+                    print(" - Receiving message from server.", flush=True)
+                    message = websocket.recv()
+                    message = message.decode("utf-8") if isinstance(message, bytes) else message
+                    # drop the newline character
+                    if message.endswith("\n"):
+                        message = message[:-1]
+
+                    print(
+                        f"   - Asserting received message '{message}' is the same as the expected message '{expected}'",
+                        flush=True,
+                    )
+                    assert message == expected
+
+                print(" - Sending message 'Yes' to server.", flush=True)
+                websocket.send("Yes")
+
+        print("Test passed.", flush=True)
+
+    @pytest.mark.skipif(skip_openai, reason="websockets module is not available or openai module is not available")
```

Comment:
```suggestion
    @pytest.mark.skipif(skip_openai, reason="websockets module is not available")
```

Reply:
actually, the message is correct as it is but it is a bit confusing. I'll refactor it a bit.